pyiron_base/jobs/flex/pythonfunctioncontainer.py

```diff
@@ -64,21 +66,28 @@ def __call__(self, *args, **kwargs):
     def to_hdf(self, hdf=None, group_name=None):
         super().to_hdf(hdf=hdf, group_name=group_name)
         self.project_hdf5["function"] = np.void(cloudpickle.dumps(self._function))
+        self.project_hdf5["_mangle_name_on_save"] = self._mangle_name_on_save
```

Comment:
Is it necessary to store this part in the HDF5 file? Basically the `to_hdf()` function is called as part of the save function and when the save function is called the action - in this case renaming the job - is already executed. Or to ask it the other way around when is it required that the property is stored in the HDF5 file?

Reply:
Yes, it is strictly necessary. Without saving this, the data is lost on re-load and you wind up with weird job renaming. In the following, _without_ saving the flag, the final job table does not contain `"my_job"` (but rather `"plus_one{hash}"` and `reloaded` is None (instead of a job):

```python
from pyiron_base import Project, JOB_CLASS_DICT

from pyiron_base.jobs.flex.pythonfunctioncontainer import PythonFunctionContainerJob

JOB_CLASS_DICT[PythonFunctionContainerJob.__name__] = PythonFunctionContainerJob.__module__

pr = Project("tmp")
pr.remove_jobs(silently=True, recursive=True)

assert(len(pr.job_table()) == 0)

def plus_one(x):
    return x + 1

# job = PythonFunctionContainerJob(pr, "my_job")  # Gave me an attribute error?!!?
job = pr.create.job.PythonFunctionContainerJob("my_job")
job.python_function = plus_one
job.input["x"] = 2
job._mangle_name_on_save = False

job.run()

loaded = pr.load("my_job")
loaded.run(delete_existing_job=True)

print(pr.job_table()["job"])
reloaded = pr.load("my_job")
type(reloaded)
```

Reply:
Ok, that makes sense.